src/Sqids.php

```diff
@@ -134,463 +57,187 @@ class Sqids implements SqidsInterface
         "puta",
-        "puta1n",
         "putain",
         "pute",
         "putiza",
         "puttana",
         "queca",
-        "r0mp1ba11e",
-        "r0mp1ba1le",
-        "r0mp1bal1e",
-        "r0mp1balle",
-        "r0mpiba11e",
-        "r0mpiba1le",
-        "r0mpibal1e",
-        "r0mpiballe",
-        "rand1",
         "randi",
         "rape",
-        "recch10ne",
-        "recch1one",
-        "recchi0ne",
         "recchione",
         "retard",
-        "romp1ba11e",
-        "romp1ba1le",
-        "romp1bal1e",
-        "romp1balle",
-        "rompiba11e",
-        "rompiba1le",
-        "rompibal1e",
         "rompiballe",
-        "ruff1an0",
-        "ruff1ano",
-        "ruffian0",
         "ruffiano",
-        "s1ut",
-        "sa10pe",
-        "sa1aud",
-        "sa1ope",
         "sacanagem",
-        "sal0pe",
         "salaud",
         "salope",
         "saugnapf",
-        "sb0rr0ne",
-        "sb0rra",
-        "sb0rrone",
         "sbattere",
-        "sbatters1",
         "sbattersi",
-        "sborr0ne",
         "sborra",
         "sborrone",
-        "sc0pare",
-        "sc0pata",
-        "sch1ampe",
-        "sche1se",
-        "sche1sse",
         "scheise",
         "scheisse",
         "schlampe",
-        "schwachs1nn1g",
-        "schwachs1nnig",
-        "schwachsinn1g",
         "schwachsinnig",
         "schwanz",
         "scopare",
         "scopata",
         "sexy",
-        "sh1t",
         "shit",
         "slut",
-        "sp0mp1nare",
-        "sp0mpinare",
-        "spomp1nare",
         "spompinare",
-        "str0nz0",
-        "str0nza",
-        "str0nzo",
-        "stronz0",
         "stronza",
         "stronzo",
-        "stup1d",
         "stupid",
-        "succh1am1",
-        "succh1ami",
-        "succhiam1",
         "succhiami",
         "sucker",
-        "t0pa",
         "tapette",
-        "test1c1e",
-        "test1cle",
-        "testic1e",
         "testicle",
         "tette",
         "topa",
-        "tr01a",
-        "tr0ia",
-        "tr0mbare",
-        "tr1ng1er",
-        "tr1ngler",
-        "tring1er",
         "tringler",
-        "tro1a",
         "troia",
         "trombare",
         "turd",
         "twat",
-        "vaffancu10",
-        "vaffancu1o",
-        "vaffancul0",
         "vaffanculo",
-        "vag1na",
         "vagina",
         "verdammt",
         "verga",
-        "w1chsen",
         "wank",
         "wichsen",
-        "x0ch0ta",
-        "x0chota",
         "xana",
-        "xoch0ta",
         "xochota",
-        "z0cc01a",
-        "z0cc0la",
-        "z0cco1a",
-        "z0ccola",
-        "z1z1",
-        "z1zi",
-        "ziz1",
         "zizi",
-        "zocc01a",
-        "zocc0la",
-        "zocco1a",
         "zoccola",
     ];
 
+    private const LEET = [
+        'i' => '[i1]',
+        'o' => '[o0]',
+        'l' => '[l1]',
+    ];
+
     protected MathInterface $math;
 
+    protected ?string $blocklist = null;
```

Comment:
changing the type of a protected property (from `array` to `?string` is a BC break. I don't know what is the Backward Compatiblity policy of this project (I got here because of your toot about performance improvements, by curiosity) but it might make sense to be care about this.

Reply:
You are right, I reverted to use an other property name and leave this one. Even if I don't see any reason this class would be extended. It should be final, it has an interface.

Reply:
If anyone extends the class and updates `$this->blocklist` after calling the constructor, then this change will not be used later in the id generator.
But this is already a misuse, as it bypasses the filter.